app/models/authenticator/ldap.rb

```diff
@@ -15,13 +15,18 @@ def ldap
       @ldap ||= ldap_bind(config[:bind_dn], config[:bind_pwd])
     end
 
+    # Unbound LDAP handle
+    def miq_ldap
+      @miq_ldap ||= MiqLdap.new(:auth => config)
+    end
```

Comment:
Do we really need 2 `MiqLdap` objects cached locally?

I wonder if simply having `ldap_bind()` return `ldap` regardless of `bind`'s success would have solved your problem.